mptt/utils.py

```diff
@@ -292,3 +295,38 @@ def get_cached_trees(queryset):
             current_path.append(obj)
 
     return top_nodes
+
+
+def get_random_string(length):
+    # With combination of lower and upper case
+    return "".join(random.choice(string.ascii_letters) for i in range(length))
+
+
+def get_default_index(model_cls):
+    return Index(
+        fields=(model_cls._mptt_meta.tree_id_attr, model_cls._mptt_meta.left_attr),
+        name=f"{get_random_string(12)}_default_mptt_idx",
```

Comment:
`get_random_string` already exists in `django.utils.crypto`. But I think it's not a good fit here. Won't this regenerate migrations over and over?

Reply:
Interesting... i do the migrations for the test app and no `AddIndex` Object is part of the migrations. So currently there are no indexes build from...

But back to the naming problem. The name of an index is limited by 30 characters and needs to be unique. The common way from django is to use [name='%(app_label)s_%(class)s_default_mptt_index')](https://docs.djangoproject.com/en/4.2/ref/models/indexes/#django.db.models.Index.name). But in our test app there will be names which are longer than 30 characters. So i made the workaround with a random string (12 chars) followed by the '_default_mptt_idx'. 

So we got two problems here. 

* No Index objects are part of migrations. For now i don't know why.
* dirty index naming; maybe they will reproduce migrations over and over.

Reply:
Yeah, I think Django has a way of creating reproducible alphanumeric suffixes for some of the database object names where it needs them. I'm not sure how it works (I didn't do any research) but I have seen some of those names in dumps etc.

> Interesting... i do the migrations for the test app and no AddIndex Object is part of the migrations. So currently there are no indexes build from...

Does that mean that the indexes themselves aren't actually created? I'm sorry for the dumb question. 

(I have marked the django-mptt package as unmaintained since maintaining it was honestly thankless, so I'm sorry for not putting more energy into the review. I hope you understand where I'm coming from.)

Reply:
> Does that mean that the indexes themselves aren't actually created?
Yes; the index operations such as creating altering shall be part of the generated migration files

> (I have marked the django-mptt package as unmaintained since maintaining it was honestly thankless, so I'm sorry for not putting more energy into the review. I hope you understand where I'm coming from.)

I understand that. It is very hard to maintain well. I think there are some overengineered parts such as the renaming mptt attribute stuff. That's also why we can't do this issue easily.

However; I will look again at the django-treebeard package if it is even possible to implement a simple parent foreignkey for me

Reply:
i started with a new implementation of [django-mptt2](https://github.com/jokiefer/django-mptt2) cause no other project fits my use cases fully.